src/routes/landing/components/CityEntry.js

```diff
@@ -25,17 +41,37 @@ type PropType = {|
 |}
 
 class CityEntry extends React.PureComponent<PropType> {
+  getMatchedAliases = (city: CityModel, normalizedFilter: string): Array<CityModel> => {
+    if (city.aliases && normalizedFilter.length >= 2) {
+      return Object.keys(city.aliases)
+        .filter(alias => normalize(alias).includes(normalizedFilter))
+    }
+    return []
+  }
+
   navigateToDashboard = () => {
     this.props.navigateToDashboard(this.props.city)
   }
 
   render () {
-    const { city, theme } = this.props
+    const { city, theme, filterText } = this.props
+    const normalizedFilter = normalize(filterText)
+    const aliases = this.getMatchedAliases(city, normalizedFilter)
     return (
-        <CityListItem onPress={this.navigateToDashboard}
-                      underlayColor={this.props.theme.colors.backgroundAccentColor}>
-          <Label theme={theme}>{city.name}</Label>
-        </CityListItem>
+      <CityListItem onPress={this.navigateToDashboard}
+                    underlayColor={theme.colors.backgroundAccentColor}>
+        <View>
+          <Label theme={theme} searchWords={[filterText]} textToHighlight={city.name} sanitize={normalize}
+                 highlightStyle={{ backgroundColor: 'yellow' }} />
+          {aliases.length > 0 && <Aliases>
+            {aliases.map(
+              (alias, index) => <AliasLabel key={alias} theme={theme} searchWords={[filterText]}
```

Comment:
Maybe you could extract this to a separate function (or move it together with getMachtingAliases).